cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -112,11 +112,14 @@ func (c *Controller) getSpcResource(key string) (*apis.StoragePoolClaim, error)
 		// The SPC resource may no longer exist, in which case we stop
 		// processing.
 		if errors.IsNotFound(err) {
+			
 			runtime.HandleError(fmt.Errorf("spcGot '%s' in work queue no longer exists", key))
-			return nil, err
 		}
-
-		return nil, err
+		// No need to return error to caller as we still want to fire the delete handler
+		// using the spc key(name)
+		// If error is returned the caller function will return without calling the spcEventHandler
+		// function that invokes business logic for pool deletion
+		return nil,nil
```

Comment:
getSpcResource should not be bothered about deletion. Can we have `separation of concerns` for this logic?

Reply:
Yes... Will be adding this in backlog and will be pushed in subsequent commit.

Reply:
ok. thanks @sonasingh46